### PR TITLE
docs: Add styleguide and document error handling

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request (what was done and why):

- Adds a place to put team-wide coding decisions
- Document error handling

Changes refer to particular issues, PRs or documents:

- 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] The feature is unit-tested.
- [ ] The feature is e2e-tested.

### CODE REVIEW

File: docs/contributor/styleguide/error_handling.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Error Handling
+
+## Reflecting Errors in Status Condition Messages
+In Telemetry Manager, it is a common practice to reflect errors in status condition messages. For instance, a LogPipeline might require a client TLS certificate to authenticate against a backend. If the certificate is invalid, the error should be captured and mapped to a corresponding reason/message in the LogPipeline status.
```
Comment (NHingerl) [inline docs/contributor/styleguide/error_handling.md:4]:
```suggestion
In Telemetry Manager, it is a common practice to reflect errors in status condition messages. For example, a LogPipeline might need a client TLS certificate to authenticate against a backend. If the certificate is invalid, the error should be captured and mapped to a corresponding reason or message in the LogPipeline status.
```

File: docs/contributor/styleguide/error_handling.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+### Defining and Using Sentinel Errors
+When an error occurs, the package that returns it (e.g., a TLS validator) should define a sentinel error that the calling code can check. For example:
```
Comment (NHingerl) [inline docs/contributor/styleguide/error_handling.md:7]:
```suggestion
When an error occurs, the package that returns it (for example, a TLS validator) should define a sentinel error that the calling code can check. For example:
```

File: docs/contributor/styleguide/error_handling.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+```go
+import "errors"
+
+var ErrCertDecodeFailed = errors.New("failed to decode PEM block containing certificate")
+```
+
+### Creating Custom Error Types for Additional Context
+If additional context is necessary, define a custom error type. For example, in the following code snippet `CertExpiredError` defines an `Expiry` field that can be used by the calling code:
```
Comment (NHingerl) [inline docs/contributor/styleguide/error_handling.md:16]:
```suggestion
If additional context is necessary, define a custom error type. For example, in the following code snippet, `CertExpiredError` defines an `Expiry` field that can be used by the calling code:
```

**Review by NHingerl — COMMENTED** [review]
This is a very useful and well-written document!

**Review by shorim — APPROVED** [review]

**Review by TeodorSAP — APPROVED** [review]

**Pull request merged**